engine/server/engine/enclave_manager/enclave_pool.go

```diff
@@ -121,11 +135,18 @@ func (pool *EnclavePool) GetEnclave(
 		return nil, nil
 	}
 
+	// If there is no idle enclave in the pool returns nil
+	// for not to block the caller
+	if len(pool.idleEnclavesChan) < oneIdleEnclave {
+		return nil, nil
+	}
 	enclaveUUID, ok := <-pool.idleEnclavesChan
 	if !ok {
 		return nil, stacktrace.NewError("A new enclave can't be returned from the pool because the internal channel is closed, it shouldn't happen; this is a bug in Kurtosis")
 	}
 
+	pool.fillChan <- fill
```

Comment:
nit: Maybe add a comment here explaining why you need to do that. It's not super explicit by just reading the code

Reply:
Yes so here, you should be expecting that as soon as you push a value to this channel, it get picked by the subroutine to start a new idle enclave, right? So, if it's picked immediately, you don't need a buffered channel

Reply:
the buffered channel is for if we have concurrent requests